From f4132bdb28be53a1cb7883c6ddf1f1d0301801de Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 05 May 2019 18:41:26 +0300
Subject: [PATCH] [SW-0] gaudi: pll: raise plls to nominal frequency

Almost all PLLs are raised to their nominal frequency.
Exceptions are:
1) PCI - is still 830MHz, has not been tested enough yet
2) IF/MESH LBW clocks - are still reference (50MHz),
   special care is needed

Change-Id: Iefa0dccc0c51d3f5ed0ac5191479fab301c4d1e5
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index bb7756d..babd0c5 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -16,63 +16,52 @@
 static struct hl_pll_cfg gaudi_pll_cfg[] = {
 	{
 		.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
-//		.nr = 0, .nf = 47, .od = 1, .nb = 23,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 47, .od = 1, .nb = 23, /* 1200 */
 		.div_factor = { 0, 1, 47, -1 }
 	}, {
 		.name = "DMA_W", .base = mmDMA_W_PLL_BASE,
-//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, -1, -1 }
 	}, {
 		.name = "IF_W", .base = mmIF_W_PLL_BASE,
-//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-		.div_factor = { 0, 1, 2, 7 }
+		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
+		.div_factor = { 0, -1, 2, 7 }
 	}, {
 		.name = "IF_E", .base = mmIF_E_PLL_BASE,
-//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-		.div_factor = { 0, 1, 2, 7 }
+		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
+		.div_factor = { 0, -1, 2, 7 }
 	}, {
 		.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
-//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, -1, -1 }
 	}, {
 		.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
-//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-		.div_factor = { 0, 1, 2, 7 }
+		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
+		.div_factor = { 0, -1, 2, 7 }
 	}, {
 		.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
-//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 35, .od = 1, .nb = 17, /* 900 */
 		.div_factor = { 0, -1, -1, -1 }
 	}, {
 		.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
-//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-		.div_factor = { 0, 1, 2, 7 }
+		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
+		.div_factor = { 0, -1, 2, 7 }
 	}, {
 		.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
-//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 35, .od = 1, .nb = 17, /* 900 */
 		.div_factor = { 0, -1, -1, -1 }
 	}, {
 		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
-//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 }
 	}, {
 		.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
-//		.nr = 0, .nf = 37, .od = 1, .nb = 18,
+//		.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 */
 		.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
 		.div_factor = { 0, 0, 4, 9 }
 	}, {
 		.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
-//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 }
 	}, {
 		.name = "HBM", .base = mmPSOC_HBM_PLL_BASE,
@@ -92,13 +81,11 @@
 		.div_factor = { 0, 1, -1, -1 }
 	}, {
 		.name = "NIC2_MME", .base = mmNIC2_MME_PLL_BASE,
-//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 }
 	}, {
 		.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
-//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-		.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 }
 	},
 };
